src/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -84,590 +57,29 @@ private static IArraySortHelper<T> CreateArraySortHelper()
-                    while (pivot.CompareTo(keys[--right]) < 0) ;
-                }
-
-                if (left >= right)
-                    break;
-
-                Swap(keys, left, right);
-            }
-
-            // Put pivot in the right location.
-            Swap(keys, left, (hi - 1));
-            return left;
-        }
-
-        private static void Heapsort(T[] keys, int lo, int hi)
-        {
-            Debug.Assert(keys != null);
-            Debug.Assert(lo >= 0);
-            Debug.Assert(hi > lo);
-            Debug.Assert(hi < keys.Length);
-
-            int n = hi - lo + 1;
-            for (int i = n / 2; i >= 1; i = i - 1)
-            {
-                DownHeap(keys, i, n, lo);
-            }
-            for (int i = n; i > 1; i = i - 1)
-            {
-                Swap(keys, lo, lo + i - 1);
-                DownHeap(keys, 1, i - 1, lo);
-            }
-        }
-
-        private static void DownHeap(T[] keys, int i, int n, int lo)
-        {
-            Debug.Assert(keys != null);
-            Debug.Assert(lo >= 0);
-            Debug.Assert(lo < keys.Length);
-
-            T d = keys[lo + i - 1];
-            int child;
-            while (i <= n / 2)
-            {
-                child = 2 * i;
-                if (child < n && (keys[lo + child - 1] == null || keys[lo + child - 1].CompareTo(keys[lo + child]) < 0))
-                {
-                    child++;
-                }
-                if (keys[lo + child - 1] == null || keys[lo + child - 1].CompareTo(d) < 0)
-                    break;
-                keys[lo + i - 1] = keys[lo + child - 1];
-                i = child;
-            }
-            keys[lo + i - 1] = d;
-        }
-
-        private static void InsertionSort(T[] keys, int lo, int hi)
-        {
-            Debug.Assert(keys != null);
-            Debug.Assert(lo >= 0);
-            Debug.Assert(hi >= lo);
-            Debug.Assert(hi <= keys.Length);
-
-            int i, j;
-            T t;
-            for (i = lo; i < hi; i++)
-            {
-                j = i;
-                t = keys[i + 1];
-                while (j >= lo && (t == null || t.CompareTo(keys[j]) < 0))
-                {
-                    keys[j + 1] = keys[j];
-                    j--;
-                }
-                keys[j + 1] = t;
-            }
-        }
     }
 
-    #endregion
-
-    #region ArraySortHelper for paired key and value arrays
-
     internal interface IArraySortHelper<TKey, TValue>
     {
         void Sort(TKey[] keys, TValue[] values, int index, int length, IComparer<TKey> comparer);
     }
 
     [TypeDependencyAttribute("System.Collections.Generic.GenericArraySortHelper`2")]
-    internal class ArraySortHelper<TKey, TValue>
+    internal partial class ArraySortHelper<TKey, TValue>
         : IArraySortHelper<TKey, TValue>
     {
         private static volatile IArraySortHelper<TKey, TValue> defaultArraySortHelper;
```

Comment:
above the private declaration of defaultArraySortHelper (which as explained earlier better be renamed to s_defaultArraySortHelper) you could add the 4 line comment below taken from corert:
https://github.com/dotnet/corert/blob/256481c2afd7290600ec4af2b7ac247436233254/src/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs#L328-L331

```
        // WARNING: We allow diagnostic tools to directly inspect this member (s_defaultArraySortHelper). 
        // See https://github.com/dotnet/corert/blob/master/Documentation/design-docs/diagnostics/diagnostics-tools-contract.md for more details. 
        // Please do not change the type, the name, or the semantic usage of this member without understanding the implication for tools. 
        // Get in touch with the diagnostics team if you have questions.
```